### PR TITLE
Add Base Colours

### PR DESCRIPTION
**Pull request by AlanaNoonan**

### CODE REVIEW

**Review by ysims — COMMENTED** [review]
A couple of things

- Add blue colour 
- Set the colour automatically using the robot's name. [See the specs](https://cdn.robocup.org/hl/wp/2021/04/v-hsc_model_specification_v1.0.pdf) for info. You'll need to use `if` statements using the name in the proto to set the textures. There are examples of [if statements in proto files here](https://github.com/cyberbotics/webots/blob/master/projects/robots/robotis/darwin-op/protos/Darwin-op.proto)

File: protos/NUgus.proto [hunk 1]
```diff
@@ -120,11 +159,32 @@ PROTO NUgus [
                 children [
                   Shape {
                     appearance PBRAppearance {
+
+                      %{
+                        if fields.name.value ~= '' then
+                          -- name is supposed to be something like "red player 2" or "blue player 1"
```
Comment (ysims) [inline protos/NUgus.proto:165]: You shouldn't need to create these variables again, the `color` and `number` variables still exist from before from what I understand of how this works.

File: protos/NUgus.proto [hunk 2]
```diff
@@ -161,13 +221,33 @@ PROTO NUgus [
...
+
                         Shape {
                           appearance PBRAppearance {
+
+                            %{
+                              if fields.name.value ~= '' then
+                                -- name is supposed to be something like "red player 2" or "blue player 1"
+                                local words = {}
+                                for word in fields.name.value:gmatch("%w+") do table.insert(words, word) end
+                                local color = words[1]
```
Comment (ysims) [inline protos/NUgus.proto:233]: Same as above.

File: protos/NUgus.proto [hunk 3]
```diff
@@ -291,11 +374,30 @@ PROTO NUgus [
                 children [
                   Shape {
                     appearance PBRAppearance {
+
+                      %{
+                        if fields.name.value ~= '' then
+                          -- name is supposed to be something like "red player 2" or "blue player 1"
```
Comment (ysims) [inline protos/NUgus.proto:380]: Same as above

File: protos/NUgus.proto [hunk 4]
```diff
@@ -334,11 +436,31 @@ PROTO NUgus [
                       children [
                         Shape {
                           appearance PBRAppearance {
+
+                            %{
+                              if fields.name.value ~= '' then
+                                -- name is supposed to be something like "red player 2" or "blue player 1"
+                                local words = {}
+                                for word in fields.name.value:gmatch("%w+") do table.insert(words, word) end
```
Comment (ysims) [inline protos/NUgus.proto:444]: Same as above

File: protos/NUgus.proto [hunk 5]
```diff
@@ -548,11 +679,31 @@ PROTO NUgus [
                             children [
                               Shape {
                                 appearance PBRAppearance {
+
+                                  %{
+                                    if fields.name.value ~= '' then
+                                      -- name is supposed to be something like "red player 2" or "blue player 1"
+                                      local words = {}
+                                      for word in fields.name.value:gmatch("%w+") do table.insert(words, word) end
+                                      local color = words[1]
```
Comment (ysims) [inline protos/NUgus.proto:688]: Same as above

File: protos/NUgus.proto [hunk 6]
```diff
@@ -890,11 +1056,30 @@ PROTO NUgus [
...
```
Comment (ysims) [inline protos/NUgus.proto:1065]: As above

File: protos/NUgus.proto [hunk 7]
```diff
@@ -25,18 +26,53 @@ PROTO NUgus [
...
+
     children [
+      %{
+        if fields.name.value ~= '' then
+          -- name is supposed to be something like "red player 2" or "blue player 1"
+          local words = {}
+          for word in fields.name.value:gmatch("%w+") do table.insert(words, word) end
+          local color = words[1]
+          local number = words[3]
+          print(color, number)
```
Comment (ysims) [inline protos/NUgus.proto:38]: I think I added this print statement, sorry. Remove this before we merge.

**Review by ysims — COMMENTED** [review]
Just some cleaning up of repeated sections that I don't think need repeating.

**Review by ysims — APPROVED** [review]
LGTM (Looks Good To Me) 😄

**Review by TheCoolKidsTable — COMMENTED** [review]
LGTM!

**Pull request merged**